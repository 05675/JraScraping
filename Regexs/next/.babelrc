{
  "presets": [["next/babel", { "class-properties": { "loose": true } }]],
  "plugins": [
    ["styled-components", { "ssr": true, "displayName": true, "preprocess": false }],
    ["@babel/plugin-proposal-decorators", { "legacy": true }],
    ["inline-react-svg"],
    ["module-resolver", { "root": ["./src"], "alias": { "@src": "./src", "@assets": "./assets" } }]
  ]
}
