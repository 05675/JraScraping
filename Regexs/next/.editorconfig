# EditorConfig is awesome: https://EditorConfig.org

# top-most EditorConfig file
root = true

# Matches multiple files with brace expansion notation
[*.{ts,tsx}]
end_of_line = lf
insert_final_newline = true
charset = utf-8
