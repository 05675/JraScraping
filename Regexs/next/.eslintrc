{
  "env": {
    "es6": true,
    "node": true,
    "jest": true,
    "browser": true
  },
  "extends": [
    "airbnb-typescript",
    "airbnb/hooks",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
    "plugin:jest/recommended",
    "plugin:prettier/recommended",
    "prettier",
    "prettier/@typescript-eslint",
    "prettier/react",
    "plugin:jsdoc/recommended"
  ],
  "plugins": ["@typescript-eslint", "prettier", "react", "react-hooks", "jest", "jsdoc"],
  "rules": {
    "@typescript-eslint/explicit-function-return-type": "off",
    "@typescript-eslint/no-unused-vars": "error",
    "@typescript-eslint/unbound-method": "off",
    "jsx-a11y/label-has-associated-control": "off",
    "react/button-has-type": "off",
    "react/destructuring-assignment": "off",
    "react/jsx-props-no-spreading": "off",
    "react/prop-types": "off",
    "complexity": ["error", 10],
    "no-console": ["warn", { "allow": ["warn", "error"] }]
  },
  "settings": {
    "react": {
      "version": "detect"
    },
    "jsdoc": {
      "mode": "typescript"
    },
    "import/resolver": {
      "typescript": {} // this loads <rootdir>/tsconfig.json to eslint
    }
  },
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "overrides": [
    {
      "files": ["*.test.{ts|tsx}"],
      "rules": {
        "@typescript-eslint/no-explicit-any": "off"
      }
    }
  ]
}
