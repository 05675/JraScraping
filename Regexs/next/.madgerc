{
  "fontName": "Noto Sans JP",
  "fontSize": "12px",
  "backgroundColor": "#000000",
  "nodeShape": "box",
  "nodeStyle": "rounded",
  "nodeColor": "#c6c5fe",
  "noDependencyColor": "#cfffac",
  "cyclicNodeColor": "#ff6c60",
  "edgeColor": "#757575",
  "fileExtensions": [
    "js",
    "jsx",
    "ts",
    "tsx"
  ],
  "tsConfig": "./tsconfig.json",
  "graphVizOptions": {
    "G": {
      "rankdir": "LR"
    }
  }
}
