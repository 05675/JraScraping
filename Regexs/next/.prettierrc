{
  "trailingComma": "es5",
  "jsxBracketSameLine": true,
  "printWidth": 100,
  "tabWidth": 2,
  "jsxSingleQuote": true,
  "singleQuote": true,
  "endOfLine": "lf"
}
