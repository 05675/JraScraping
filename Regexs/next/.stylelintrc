{
  "processors": [
    [
      "stylelint-processor-styled-components",
      {
        "moduleName": "styled-components",
        "importName": "default",
        "strict": false,
        "ignoreFiles": [],
        "parserPlugins": [
          "jsx",
          ["decorators", { "decoratorsBeforeExport": true }],
          "classProperties",
          "exportExtensions",
          "functionBind",
          "functionSent"
        ]
      }
    ]
  ],
  "extends": [
    "stylelint-config-recommended",
    "stylelint-config-styled-components",
    "stylelint-config-prettier",
    "stylelint-prettier/recommended"
  ],
  "plugins": ["stylelint-prettier", "stylelint-declaration-block-no-ignored-properties"],
  "rules": {
    "prettier/prettier": true,
    "plugin/declaration-block-no-ignored-properties": true
  }
}
