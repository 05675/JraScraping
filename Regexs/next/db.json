{
  "getMyTasks": [
    {
      "id": "uuid1",
      "title": "1234",
      "summary": "1234",
      "explanation": "1234",
      "due_date": "2020-02-05",
      "emp_id": "1234",
      "status": "0",
      "created_at": "2020-02-06"
    },
    {
      "id": "uuid2",
      "title": "1234",
      "summary": "1234",
      "explanation": "1234",
      "due_date": "2020-02-05",
      "emp_id": "1234",
      "status": "0",
      "created_at": "2020-02-06"
    },
    {
      "id": "uuid3",
      "title": "1234",
      "summary": "1234",
      "explanation": "1234",
      "due_date": "2020-02-05",
      "emp_id": "1234",
      "status": "0",
      "created_at": "2020-02-06"
    },
    {
      "id": "uuid4",
      "title": "1234",
      "summary": "1234",
      "explanation": "1234",
      "due_date": "2020-02-05",
      "emp_id": "1111",
      "status": "0",
      "created_at": "2020-02-06"
    },
    {
      "id": "uuid5",
      "title": "1234",
      "summary": "1234",
      "explanation": "1234",
      "due_date": "2020-02-05",
      "emp_id": "0000",
      "status": "0",
      "created_at": "2020-02-06"
    },
    {
      "id": "uuid6",
      "title": "1234",
      "summary": "1234",
      "explanation": "1234",
      "due_date": "2020-02-05",
      "emp_id": "0000",
      "status": "0",
      "created_at": "2020-02-06"
    }
  ]
}
