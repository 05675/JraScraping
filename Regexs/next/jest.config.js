module.exports = {
  testEnvironment: 'jsdom',
  preset: 'ts-jest',
  setupFilesAfterEnv: ['<rootDir>/setupTests.ts'],
  verbose: true,
  moduleFileExtensions: ['js', 'jsx', 'ts', 'tsx'],
  testPathIgnorePatterns: ['<rootDir>/.next/', '<rootDir>/node_modules/'],
  globals: {
    'ts-jest': {
      tsConfig: '<rootDir>/tsconfig.jest.json',
    },
  },
};
