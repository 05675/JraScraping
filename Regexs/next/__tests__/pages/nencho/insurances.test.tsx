import Insuarances from '@src/pages/nencho/insurances';
import { NextPageContext } from 'next';
import React from 'react';
import renderer from 'react-test-renderer';

describe('保険料控除_入力前（従業員）', () => {
  test('snapshot', async () => {
    const props = await Insuarances.getInitialProps?.(getNextPageContext());
    if (props) {
      const content = renderer.create(<Insuarances {...props} />).toJSON();
      expect(content).toMatchSnapshot();
      return;
    }
    throw new Error('スナップショットの取得に失敗');
  });

  test('getInitialProps_正常系_返り値', async () => {
    // 期待値
    const expectdValue = { firm: '弥生生命', kind: '生命保険（一般）' };
    const props = await Insuarances.getInitialProps?.(getNextPageContext());
    expect(props?.listItems[0].firm).toBe(expectdValue.firm);
    expect(props?.listItems[0].kind).toBe(expectdValue.kind);
  });
});

const getNextPageContext = () => {
  const nextPageMock = jest.fn();
  const ctx: NextPageContext = new nextPageMock();
  return ctx;
};
