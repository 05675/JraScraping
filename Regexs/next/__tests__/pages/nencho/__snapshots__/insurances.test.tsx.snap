// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`保険料控除_入力前（従業員） snapshot 1`] = `
<div>
  <section
    className="section"
  >
    <div
      className="container"
    >
      <header
        className="sc-Axmtr iJkvvF"
      >
        <div
          className="productNameHeader"
        >
          <div
            className="arrow"
          >
            ←
          </div>
          <div
            className="backLink"
          >
            2020年分年末…
          </div>
          <div
            className="productTitle"
          >
            弥生給与
          </div>
          <div
            className="rightSideLink"
          />
        </div>
        <div
          className="menuNameHeader"
        >
          保険料控除
        </div>
      </header>
      <div
        className="sc-AxhCb hNncAC"
      >
        <div
          className="sc-AxhUy crScAy"
        >
          <div>
            生命保険（一般）
          </div>
          <div
            className="sc-AxgMl hbAoqV"
          >
            弥生生命
          </div>
        </div>
        <div
          className="sc-AxhUy sc-AxheI bTOMuE"
        >
          →
        </div>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="sc-AxmLO BTnWA"
        >
          <button
            className="sc-AxjAm kdprGj"
          >
            +　新しい保険を追加する
          </button>
        </div>
      </form>
    </div>
  </section>
</div>
`;
